rafx-framework/src/visibility/view_frustum_arc.rs

```diff
@@ -100,13 +101,18 @@ impl ViewFrustumArc {
         self
     }
 
-    pub fn query_visibility(&mut self) -> RafxResult<RwLockReadGuard<VisibilityQuery>> {
+    pub fn query_visibility(
+        &mut self,
+        visibility_config: &VisibilityConfig,
+    ) -> RafxResult<RwLockReadGuard<VisibilityQuery>> {
         let mut results = self.inner.visibility_query.write();
 
         results.objects.clear();
         results.volumes.clear();
 
-        self.inner.visibility_world.update();
+        if visibility_config.enable_visibility_update {
```

Comment:
I think I would probably use this to skip the `clear` on the 2 lines above, and then skip lines 119 through 129 (the actual calls to `query_visibility`). The issue with skipping the updates to the visibility world is that I think you'll be piling up channel updates (a memory leak, of sorts).